platform/posix/transport/utest/reconnect_utest.c

```diff
@@ -0,0 +1,128 @@
+#include <string.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include "unity.h"
+
+/* Include paths for public enums, structures, and macros. */
+#include "transport_reconnect.h"
+
+#include "mock_unistd_api.h"
+#include "mock_stdio_api.h"
+
+static TransportReconnectParams_t reconnectParams;
+
+/* ============================   UNITY FIXTURES ============================ */
+
+/* Called before each test method. */
+void setUp()
+{
+}
+
+/* Called after each test method. */
+void tearDown()
+{
+}
+
+/* Called at the beginning of the whole suite. */
+void suiteSetUp()
+{
+}
+
+/* Called at the end of the whole suite. */
+int suiteTearDown( int numFailures )
+{
+    return numFailures;
+}
+
+/* ========================================================================== */
+
+/**
+ * @brief A helper method to make assertions on the #TransportReconnectParams_t
+ * object upon calling #Transport_ReconnectParamsReset.
+ */
+static void verifyReconnectParamsAfterReset( void )
+{
+    TEST_ASSERT_EQUAL( 0, reconnectParams.attemptsDone );
+    TEST_ASSERT_GREATER_OR_EQUAL_INT32( INITIAL_RECONNECT_BACKOFF_SECONDS,
+                                        reconnectParams.nextJitterMax );
+    TEST_ASSERT_LESS_THAN_INT32( ( INITIAL_RECONNECT_BACKOFF_SECONDS +
+                                   MAX_JITTER_VALUE_SECONDS ),
+                                 reconnectParams.nextJitterMax );
+}
+
```

Comment:
Please add a visual break between the utility function and tests
```suggestion
/* ====================  Test Cases ========================== */
```